### PR TITLE
Add an nl option to encode() and remove caching

### PR DESCRIPTION
**Pull request by knu**
All existing and past vCard specifications agree that the line terminator should be CRLF, so let's give users an option to comply with that.

https://web.archive.org/web/20160402232011/https://www.imc.org/pdi/vcard-21.txt (vCard 2.1)
https://tools.ietf.org/html/rfc2426 (vCard 3.0)
https://tools.ietf.org/html/rfc6350 (vCard 4.0)

### CODE REVIEW

File: lib/vcard/dirinfo.rb [hunk 1]
```diff
@@ -242,11 +242,14 @@ def delete(field)
...
+
     # The string encoding of the DirectoryInfo. See Field#encode for information
-    # about the width parameter.
-    def encode(width=75)
+    # about the parameters.
+    def encode(width = 75, nl: LF)
       unless @string
-        @string = @fields.collect { |f| f.encode(width) }.join ""
+        options = [width, nl: nl]
+        @string = @fields.collect { |f| f.encode(*options) }.join
```
Comment (brendon) [inline lib/vcard/dirinfo.rb:252]: How about removing the `options` variable and just do:

```
@string = @fields.collect { |f| f.encode(width, nl: nl) }.join
```

?

**Comment by brendon** [comment]
Great idea! Thanks for the PR. Just one thing to look at :)

File: lib/vcard/dirinfo.rb (hunk 1 above)
Comment (knu) [reply lib/vcard/dirinfo.rb]: Done!

**Comment by knu** [comment]
I also removed caching from DirectoryInfo#encode.  It now has many options and caching could be a pitfall while you wouldn't normally encode a card more than once.

**Comment by brendon** [comment]
Excellent, Travis is taking a long time to run these days. If you notice that this is all green can you give me a ping and I'll merge it.

**Review by brendon — APPROVED** [review]

**Comment by brendon** [comment]
Thanks @knu. You're a pleasure to work with :)

**Comment by knu** [comment]
Thank you too!  This was the last feature I had in mind to roll into the next release.

**Pull request merged**